File: app/src/main/java/com/recursia/popularmovies/domain/FavoriteScreenInteractorImpl.java
package com.recursia.popularmovies.domain;

import com.recursia.popularmovies.domain.models.Movie;

import java.util.List;

import io.reactivex.Flowable;

public class FavoriteScreenInteractorImpl implements FavoriteScreenInteractor {

    private final MoviesRepository moviesRepository;

    public FavoriteScreenInteractorImpl(MoviesRepository moviesRepository) {
        this.moviesRepository = moviesRepository;
    }

    @Override
    public Flowable<List<Movie>> getAllFavoriteMovies() {
        return moviesRepository.getAllFavoriteMovies();
    }
}

File: app/src/main/java/com/recursia/popularmovies/domain/FavoriteScreenInteractorImpl.kt
package com.recursia.popularmovies.domain

import com.recursia.popularmovies.domain.models.Movie

import io.reactivex.Flowable

class FavoriteScreenInteractorImpl(private val moviesRepository: MoviesRepository) : FavoriteScreenInteractor {

    override val allFavoriteMovies: Flowable<List<Movie>>
        get() = moviesRepository.allFavoriteMovies

}
